project/SparkBuild.scala

```diff
@@ -90,8 +90,6 @@ object BuildCommons {
   // Google Protobuf version used for generating the protobuf.
   // SPARK-41247: needs to be consistent with `protobuf.version` in `pom.xml`.
   val protoVersion = "4.29.3"
```

Comment:
https://github.com/apache/spark/blob/93e56c26acec8455ec532fcedea9e22fd05a288a/project/SparkBuild.scala#L1113-L1135

Due to the `sbtpomreader.PomReaderPlugin` being disabled in `OldDeps`, if a similar approach is used to obtain `protoVersion`, it will fail to retrieve it at line 1132. Therefore, in the current PR, only the retrieval of `grpcVersion` has been fixed. I will address the retrieval method for `protoVersion` once I find a suitable approach.